src/comments.js

```diff
@@ -0,0 +1,88 @@
+import dialogPolyfill from 'dialog-polyfill';
+import {
+  getComments,
+  postComments,
+} from './api';
+
+import { displayCommentsCounter } from './counters';
+
+const dialog = document.querySelector('#dialog');
+dialogPolyfill.registerDialog(dialog);
+
+const commentsList = document.querySelector('.comments-list');
+
+const displayError = () => {
+  const commentMessage = document.createElement('li');
+  commentMessage.className = 'no-comments';
+  commentMessage.textContent = 'No Comments';
+  commentsList.append(commentMessage);
+};
+
+const displayComment = (comment) => {
+  const commentElement = document.createElement('li');
+  commentElement.className = 'comment';
+  commentElement.textContent = `${comment.creation_date} ${comment.username} : ${comment.comment}`;
+  commentsList.append(commentElement);
+};
+
+const displayComments = async (showId) => {
+  commentsList.innerHTML = '';
+
+  try {
+    const comments = await getComments(showId);
+    comments.forEach((comment) => displayComment(comment));
+    displayCommentsCounter(dialog, commentsList);
+  } catch (error) {
+    displayError();
+  }
+};
+
+const validateComment = (name, comment) => name.value !== '' && comment.value !== '';
+
+const inputComment = async () => {
+  const nameInput = document.querySelector('#name');
+  const commentInput = document.querySelector('#comment-input');
+  if (validateComment(nameInput, commentInput)) {
+    const commentObj = {
+      item_id: `${commentsList.id}`,
+      username: `${nameInput.value.trim()}`,
+      comment: `${commentInput.value.trim()}`,
+    };
+    nameInput.value = '';
+    commentInput.value = '';
+    await postComments(commentObj);
+    displayComments(commentsList.id);
+  }
+};
+
+const popup = (e) => {
+  commentsList.innerHTML = '';
+  const showElement = e.target.parentNode.parentNode;
```

Comment:
- |OPTIONAL] It would be good to have the image of the movie at the top of the popup. This brings more details to the item like shown in the design wireframe

![Screenshot from 2021-10-15 14-20-27](https://user-images.githubusercontent.com/40503704/137493603-3e356338-2546-462c-b58e-da0b2aee5d61.png)